deltacat/compute/compactor/model/round_completion_info.py

```diff
@@ -6,6 +6,22 @@
 from typing import Any, Dict, Optional, Union
 
 
+class HighWaterMark(dict):
+    """
+    Inherit from dict to make it easy for serialization/deserialization.
+    Keep both partition locator and high watermark as a tuple to be persisted in the rcf
+    """
+
+    def append(self, partition_locator: PartitionLocator, high_watermark: int):
+        self[partition_locator.canonical_string()] = (partition_locator, high_watermark)
```

Comment:
Although I recommended `append` initially, looking at this again I think `set` is a better natural fit here alongside `get`. Also, since the class is already named `HighWatermark` it may be more descriptive to call the integer here `delta_stream_position`?

```suggestion
    def set(self, partition_locator: PartitionLocator, delta_stream_position: int):
        self[partition_locator.canonical_string()] = (partition_locator, delta_stream_position)
```